hot-reload-agent/src/main/kotlin/org/jetbrains/compose/reload/agent/reload.kt

```diff
@@ -109,10 +113,33 @@ internal fun reload(
         ClassDefinition(originalClass, transformed)
     }
 
+    val changedResources = pendingChanges.keys.filter { !it.isClass() && it.isFile }
+
+    if (changedResources.isNotEmpty()) {
+        cleanResourceCache()
+    }
+
     instrumentation.redefineClasses(*definitions.toTypedArray())
-    return redefineRuntimeInfo().get().mapLeft { redefinition ->
+    return redefineRuntimeInfo(changedResources).get().mapLeft { redefinition ->
         val reload = Reload(reloadRequestId, definitions, redefinition)
         reinitializeStaticsIfNecessary(reload)
         reload
     }
 }
+
+private fun cleanResourceCache(classId: ClassId, cleanMethod: MethodId, resourceCacheType: String) {
```

Comment:
Similarly to handling statics, this code should likely live in its own source file. 

I can totally see that this can be an extension working in 'invokeAfterHotReload'